agents/gce/fence_gce.py

```diff
@@ -38,50 +56,141 @@ def get_power_status(conn, options):
+	try:
+		logging.info('Issuing reset of %s in zone %s' % (options["--plug"], options["--zone"]))
+		operation = conn.instances().reset(
+				project=options["--project"],
+				zone=options["--zone"],
+				instance=options["--plug"]).execute()
+		wait_for_operation(conn, options["--project"], options["--zone"], operation)
+		logging.info('Reset of %s in zone %s complete' % (options["--plug"], options["--zone"]))
+		return True
+	except Exception as err:
+		logging.error("Failed: power_cycle: {}".format(str(err)))
+		return False
+
+
+def get_instance(conn, project, zone, instance):
+	request = conn.instances().get(
+			project=project, zone=zone, instance=instance)
+	return request.execute()
+
+
+def get_zone(conn, project, instance):
+	request = conn.instances().aggregatedList(project=project)
+	while request is not None:
+		response = request.execute()
+		zones = response.get('items', {})
+		for zone in zones.values():
+			for inst in zone.get('instances', []):
+				if inst['name'] == instance:
+					return inst['zone'].split("/")[-1]
+		request = conn.instances().aggregatedList_next(
+				previous_request=request, previous_response=response)
+	raise Exception("Unable to find instance %s" % (instance))
+
+
+def get_metadata(metadata_key, params=None, timeout=None):
+	"""Performs a GET request with the metadata headers.
+
+	Args:
+		metadata_key: string, the metadata to perform a GET request on.
+		params: dictionary, the query parameters in the GET request.
+		timeout: int, timeout in seconds for metadata requests.
+
+	Returns:
+		HTTP response from the GET request.
+
+	Raises:
+		urlerror.HTTPError: raises when the GET request fails.
+	"""
+	timeout = timeout or 60
+	metadata_url = os.path.join(METADATA_SERVER, metadata_key)
+	params = urlparse.urlencode(params or {})
+	url = '%s?%s' % (metadata_url, params)
+	request = urlrequest.Request(url, headers=METADATA_HEADERS)
+	request_opener = urlrequest.build_opener(urlrequest.ProxyHandler({}))
+	return request_opener.open(request, timeout=timeout * 1.1).read()
+
+
 def define_new_opts():
 	all_opt["zone"] = {
 		"getopt" : ":",
 		"longopt" : "zone",
 		"help" : "--zone=[name]                  Zone, e.g. us-central1-b",
 		"shortdesc" : "Zone.",
-		"required" : "1",
+		"required" : "0",
 		"order" : 2
 	}
 	all_opt["project"] = {
 		"getopt" : ":",
 		"longopt" : "project",
 		"help" : "--project=[name]               Project ID",
 		"shortdesc" : "Project ID.",
-		"required" : "1",
+		"required" : "0",
 		"order" : 3
 	}
+	all_opt["stackdriver-logging"] = {
+		"getopt" : "",
+		"longopt" : "stackdriver-logging",
+		"help" : "--stackdriver-logging		Enable Logging to Stackdriver",
+		"shortdesc" : "Stackdriver-logging support.",
+		"longdesc" : "If enabled IP failover logs will be posted to stackdriver logging.",
+		"required" : "0",
+		"order" : 4
+	}
+
 
 def main():
 	conn = None
 
-	device_opt = ["port", "no_password", "zone", "project"]
+	hostname = platform.node()
```

Comment:
You should use options["--plug"] instead of this, to avoid incosistency (and users might use different instance name and hostname).

Reply:
hostname is just for logging to stackdriver

Reply:
Yeah, but I dont think it's necessary to import platform and define a new variable here when you can just use options["--plug"] on line 214 instead.

Reply:
hostname is only used by stackdriver logs, if I understood correctly, the action `-o list` does't require `--plug`, so if I remove the hostname, the stackdriver won't be initialized correctly if no `--plug` is provided.

Reply:
I guess you could skip stackdriver logging for those calls (as local logging should be sufficient when not fencing or checking the status of an instance).

Reply:
agreed, I'll update the PR